marathon/src/main/scala/io/buoyant/marathon/v2/AppIdNamer.scala

```diff
@@ -0,0 +1,119 @@
+package io.buoyant.marathon.v2
+
+import com.twitter.finagle.{Addr, Name, Namer, NameTree, Path}
+import com.twitter.finagle.util.DefaultTimer
+import com.twitter.util._
+import java.net.SocketAddress
+
+object AppIdNamer {
+  object Closed extends Throwable
+  val log = Api.log
+}
+
+class AppIdNamer(
+  api: Api,
+  prefix: Path,
+  ttl: Duration,
+  timer: Timer = DefaultTimer.twitter
+) extends Namer {
+
+  import AppIdNamer._
+
+  private[this] implicit val _timer = timer
+
+  /**
+   * Accepts names in the form:
+   *   /<app-id>/residual/path
+   *
+   * and attempts to bind an Addr by resolving named endpoint from the
+   * Marathon master.
+   */
+  def lookup(path: Path): Activity[NameTree[Name]] = path match {
+    case Path.Utf8(app, _*) =>
+      val id = prefix ++ Path.Utf8(app)
+      val residual = path.drop(1)
+      // Marathon appends a leading '/' to app names that don't have it.
+      val appId = if (app(0) == '/') {
+        app
+      } else {
+        s"/$app"
+      }
+      log.debug("marathon lookup: %s %s %s", id.show, path.show, appId)
+      appsActivity.map {
+        case apps if apps(appId) =>
+          val addr = watchAddr(appId)
+          log.debug("marathon app found: (%s) appId: %s addr: %s", apps.mkString(", "), appId, addr)
+          NameTree.Leaf(Name.Bound(addr, id, residual))
+        case apps =>
+          log.debug("marathon app not found: (%s) appId: %s", apps.mkString(", "), appId)
```

Comment:
We have a ticket elsewhere for this, but it's worth preempting it here: Let's use Trace.record, which will save us a lot of time in the longterm.  E.g.
```scala
Trace.recordBinary("marathon.appId", appId)
...
Trace.record("marathon.found", true)
...
Trace.record("marathon.found", false)
```

It's probably not useful to log `addr` here (since it changes on its own, and this will just be some memory address here)